registration/tests/models.py

```diff
@@ -471,17 +474,332 @@ def test_resend_activation_email_expired_user(self):
         new_user.save()
 
-        profile = RegistrationProfile.objects.get(user=new_user)
+        profile = self.registration_profile.objects.get(user=new_user)
         self.assertTrue(profile.activation_key_expired())
 
-        self.assertFalse(RegistrationProfile.objects.resend_activation_mail(
+        self.assertFalse(self.registration_profile.objects.resend_activation_mail(
             email=self.user_info['email'],
             site=Site.objects.get_current(),
         ))
         self.assertEqual(len(mail.outbox), 0)
+
+
+class SupervisedRegistrationModelTests(RegistrationModelTests):
+    """
+    Test the model and manager used in the admin_approval backend.
+
+    """
+
+    user_info = {'username': 'alice',
+                 'password': 'swordfish',
+                 'email': 'alice@example.com'}
+
+    registration_profile = SupervisedRegistrationProfile
+
+    def setUp(self):
+        self.old_activation = getattr(settings,
+                                      'ACCOUNT_ACTIVATION_DAYS', None)
+        self.old_reg_email = getattr(settings,
+                                     'REGISTRATION_DEFAULT_FROM_EMAIL', None)
+        self.old_email_html = getattr(settings,
+                                      'REGISTRATION_EMAIL_HTML', None)
+        self.old_django_email = getattr(settings,
+                                        'DEFAULT_FROM_EMAIL', None)
+        self.old_admins = getattr(settings,
+                                  'ADMINS', None)
+
+        settings.ACCOUNT_ACTIVATION_DAYS = 7
+        settings.REGISTRATION_DEFAULT_FROM_EMAIL = 'registration@email.com'
+        settings.REGISTRATION_EMAIL_HTML = True
+        settings.DEFAULT_FROM_EMAIL = 'django@email.com'
+        settings.ADMINS = (
+            ('T-Rex', 'admin1@iamtrex.com'),
+            ('Flea', 'admin2@iamaflea.com')
+        )
+
+    def tearDown(self):
+        settings.ACCOUNT_ACTIVATION_DAYS = self.old_activation
+        settings.REGISTRATION_DEFAULT_FROM_EMAIL = self.old_reg_email
+        settings.REGISTRATION_EMAIL_HTML = self.old_email_html
+        settings.DEFAULT_FROM_EMAIL = self.old_django_email
+
+    def test_valid_activation(self):
+        """
+        Activating a user within the permitted window makes the
+        account active, and resets the activation key.
+
+        """
+        new_user = self.registration_profile.objects.create_inactive_user(
+            site=Site.objects.get_current(), **self.user_info)
+        profile = self.registration_profile.objects.get(user=new_user)
+        activated = (self.registration_profile.objects
+                     .activate_user(profile.activation_key))
+
+        self.failUnless(isinstance(activated, UserModel()))
+        self.assertEqual(activated.id, new_user.id)
+        self.failIf(activated.is_active)
+
+        profile = self.registration_profile.objects.get(user=new_user)
+        self.assertTrue(profile.activated)
+
+    def test_valid_activation_with_profile(self):
```

Comment:
Same with ``test_valid_activation()`` above, ``new_user.is_active`` should be ``False``